File: example/authenticate.dart
import 'package:pawnote/pawnote.dart';

void main() async {
  final client = await Authenticator.withStudentCredentials(
    loginPortalUrl: 'https://demo.index-education.net/pronote',
    username: 'demonstration',
    password: 'pronotevs',
    deviceUuid: 'my-device-uuid'
  );

  print("Authenticated as '${client.username}' on instance '${client.instanceRootUri}' with token '${client.nextTimeToken}'.");
}

File: examples/javascript/authenticate.ts
import { WebSpace, authenticate_with_credentials } from "pawnote";

const session = await authenticate_with_credentials(
  "https://demo.index-education.net/pronote",
  WebSpace.Students,
  "demonstration",
  "pronotevs",
  "device-uuid",
);

const session_information = session.information;
console.log(session_information.id);